### PR TITLE
Remove ObjCThemis.xcodeproj

### PR DESCRIPTION
**Pull request by ilammy**
The idea behind building `objcthemis.framework` (added in #604) has been to unify import syntax between Carthage and CocoaPods. Unfortunately, it turned out to be a mistake. `objcthemis.framework` does not work without `themis.framework` being present alongside it because of how module resolution works. Despite `objcthemis.framework` providing the same `themis` module as `themis.framework`, the compiler will look for a framework named `themis.framework` when resolving `import themis`.

Moreover, the original issue that `objcthemis.framework` has been called to rectify can be resolved more elegantly by importing the module:

```objective-c
@import themis;
```

which work well with `themis.framework` in both Carthage and CocoaPods.

Since `objcthemis.framework` does not bring any value, remove it. Move all new things added to ObjCThemis.xcodeproj into Themis.xcodeproj (such as testing Swift 4 vs 5). Remove the import warning. Now Carthage will build only one framework: `themis.framework` from Themis.xcodeproj.

I am sorry for the trouble and confusion of this fizzled migration 🙏 😣

## Checklist

- [X] Change is covered by automated tests
- [X] The [coding guidelines] are followed
- [X] Example projects and code samples are up-to-date (#703)
- [X] Changelog is updated

[coding guidelines]: https://github.com/cossacklabs/themis/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +28,16 @@ _Code:_
 
   - Updated Objective-C examples (iOS and macOS, Carthage and CocoaPods) to showcase usage of the newest Secure Cell API: generating symmetric keys and using Secure Cell with Passphrase ([#688](https://github.com/cossacklabs/themis/pull/688)) and to use latest Themis 0.13.1 ([#701](https://github.com/cossacklabs/themis/pull/701), [#703](https://github.com/cossacklabs/themis/pull/703)).
 
+  - **Breaking changes**
+
+    - <a id="0.13.2-revert-objcthemis-rename">ObjCThemis framework built by Carthage is now called `themis.framework` once again</a> ([#704](https://github.com/cossacklabs/themis/pull/704)).
+
+      [ObjCThemis 0.13.0](#0.13.0-objcthemis-rename) has initiated renaming of the framework produced by Carthage into `objcthemis.framework` from its historical name `themis.framework`. This decision has been a mistake. More information is available in the pull request linked above.
```
Comment (vixentael) [inline CHANGELOG.md:35]: > This decision has been a mistake.

harsh 😂

**Comment by vixentael** [comment]
If unit tests are passing, then we've tested all use cases (🤞🤞🤞🤞)

**Comment by ilammy** [comment]
Turns out you can't easily remove a thing just like that in wonderful world of development for fruit company property. The last bunch of commits fixes up various issues that rose after an attempt to simply rename the project files. In particular, unit tests get a workaround for the issue which the original ObjCThemis.xcodeproj intended to solve.

**Pull request merged**